app/tests/src/test_generate.py

```diff
@@ -112,6 +112,34 @@ def test_generate(monkeypatch):
     assert generate("gpt-4o", PROMPT, "some query") == expected_response
 
 
+def test_generate_with_history(monkeypatch):
+    monkeypatch.setattr("src.generate.completion", mock_completion.mock_completion)
+    history = [
+        {"content": "some query", "role": "user"},
+        {"role": "assistant", "content": "<div> answer</div>"},
+        {"content": "some other query", "role": "user"},
+    ]
+    expected_response = (
+        'Called gpt-4o with [{"content": "'
+        + PROMPT
+        + '", "role": "system"}, {"content": "Use the following context to answer'
+        + ' the question: context", "role": "system"},'
+        + ' {"content": "some query", "role": "user"}, {"content": "<div> answer</div>",'
+        + ' "role": "assistant"}, {"content": "some other query", "role": "user"}]'
```

Comment:
Make this easier to read:
```suggestion
        'Called gpt-4o with [{"content": "' + PROMPT + '", "role": "system"}, "
        '{"content": "Use the following context to answer the question: context", "role": "system"}, '
        '{"content": "some query", "role": "user"}, '
        '{"content": "<div> answer</div>", "role": "assistant"}, '
        '{"content": "some other query", "role": "user"}]'
```

Reply:
this auto formats to 
```
'Called gpt-4o with [{"content": "'
        + PROMPT
        + '", "role": "system"}, "'
        + '{"content": "Use the following context to answer the question: context", "role": "system"}, '
        '{"content": "some query", "role": "user"}, '
        '{"content": "<div> answer</div>", "role": "assistant"}, '
        '{"content": "some other query", "role": "user"}]'
```

Reply:
You can use an f-string for the first line to remove the `+`s.